src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -128,6 +131,13 @@
     private final boolean dlsFlsEnabled;
     private DynamicConfigModel dcm;
 
+    private static final Pattern dnfofPatterns = Pattern.compile(
+            "indices:data/read/*" + "|" +
```

Comment:
The goal of using regex pattern is that there are common prefixes (`index:`, `admin/`) that matcher can optimize, the current pattern is not much better than the original code.

Reply:
pattern optimized